### ISSUE TITLE
BUG - sev-3 - iOS - Unable to select links in messages with screen reader on

### ISSUE THREAD

**Issue by TKDickson** [department-of-veterans-affairs/va-mobile-app#9210]
Everything that follows is a copy-paste of #7615. That was closed as fixed/a duplicate, but the "ticket that was actually fixed" wasn't linked in there - @Sparowhawk can you add when you're back?

So - unsure if this was fixed and quickly regressed, or never fixed, or what's going on. But it's definitely a problem in today's develop build.

<img width="363" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeFV5Qmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--ce72213e9a240884266f84593571b8941b275427/image.png" alt="image.png" />

## What happened?

In the process of creating clickable links in the body of secure messages we uncovered a limitation in react native on iOS that doesn't allow the links to be select when voiceover is on.

## Specs:

- Device: ipad mini
- OS: iOS
- User Account: N/A ( use demo mode for message with links)
- Accessibility State: voiceover enabled

## Steps to Reproduce

- with voiceover on navigate to a secure message with links in the body of the message
- attempt to explore by touch, or swipe to a link
- observe links are not selectable

## Desired behavior

- links should be selectable

## Bug Severity - BE SURE TO ADD THE SEVERITY LABEL
 
See [Bug Tracking](https://department-of-veterans-affairs.github.io/va-mobile-app/docs/QA#issue-severity) for details on severity levels

- Impact: Low 
- Frequency: Low 

- 3 - Low

## Linked to Story

https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/5712

## Screen shot(s) and additional information

<details>
  <summary>Full JSON response for services related to issue (expand/collapse)</summary>
  

</details>

## Ticket Checklist
- [ ] Steps to reproduce are defined
- [ ] Desired behavior is added
- [ ] Labels added (front-end, back-end, relevant feature, accessibility, etc)
- [ ] Estimate of 1 added (for front-end tickets)
- [ ] Added either to the relevant feature epic (if found during new feature implementation), or the [bug epic](https://app.zenhub.com/workspaces/va-mobile-product-view-610035bc5395bb000e62e529/issues/gh/department-of-veterans-affairs/va-mobile-app/5367)

**Comment by rbontrager** [comment]
@Sparowhawk This still isn't working.  I'm tried to run it on an iPhone 13 running 17.5 and and a iPhone running 16.3 and in both instances I was unable to select links in the messages with voiceOver on.

**Comment by rbontrager** [comment]
@Sparowhawk The current solution of having to swipe through each word individually is pretty horrible for screen readers.  Is there another solution for this? @brea11y (for notification purposes on this thread)

**Comment by Sparowhawk** [comment]
@brea11y I am going to try to group all of the text outside of links together so it will be read/selectable as <Text> <Link> <Text> instead of <Text> x however many <Link>

That way all of the text is navigable, and it reduces the number of elements as much as possible

**Comment by brea11y** [comment]
That sounds like a great plan! Thank you!

**Comment by rbontrager** [comment]
@Sparowhawk Voiceover is saying there is a blank space after the message when it shouldn't be.  I think it may be caused by the links. 

[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOFRrQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--34a88309b6281060d83339a7ec5cd15ff20bb9fb/RPReplay_Final1726670243.mp4)

**Comment by rbontrager** [comment]
@Sparowhawk Was able to make some custom messages and there are a couple issues happening when you the message is something like <link 1> enter key <link 2> or <link 1> enter key <text> enter key <link 2>.  The <text> is right aligning and voice over is again picking up blank spaces. 

[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBdzNyQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--2a4511466a317527a2c964e6317c214fa6186721/RPReplay_Final1726766194.mp4)

**Comment by Sparowhawk** [comment]
the spacing being accessible/selectable should be fixed, the issue with the test aligning on the right like that is due to the new way of rendering it.

Instead of the old way where it was

<TextView>
   {Colleaction of TextViews and Links}
</TextView>

It is now textviews and links as needed with no parent textview that forces the wrapping.

<TextView>
<Link>

This is an unintended consequence, but it is the best that can be done.

**Comment by rbontrager** [comment]
@wavelaurenrussell Bringing you into this conversation.  Dylan's solution to the voiceOver issue causes some text in messages to be right/weirdly aligned (you can see it in my video above with the 'or') and I just want to make sure that this or isn't acceptable from a UX perspective.

**Comment by wavelaurenrussell** [comment]
Thanks! A couple assumptions I am making when coming into this at this point: 

1. The initial objective for the new solution is to list each link on their own line, opposed to the large block of text (inline cluster) that they were displaying as before? 
2. The large gap of space between is/will be fixed in an upcoming merge?

If the answers are Yes and Yes, then its fine. If not, then I have some follow-up q's.

**Comment by rbontrager** [comment]
@wavelaurenrussell The initial point of this ticket was to have the ability for a user to use voiceover to select links embedded in the message.  I was just testing various messages to make sure everything looks good. 

- The reason each link is on a separate line is because I made the message where I entered a link, pressed enter, wrote 'or' on the new line, pressed enter again and then entered another link (hopefully that makes sense).  And from what Dylan is saying he can't fix that with his current solution.

**Comment by rbontrager** [comment]
@wavelaurenrussell It currently does.  Dylan says he is going to look into making normal text left-aligned while keeping the a11y solution.  Moving this back to in progress while he investigates.

**Comment by Sparowhawk** [comment]
<img width="421" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMDRGQnc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--1f480a1cd9ed6f94f4a741482783276e9a797b6e/Screenshot%202024-09-26%20at%2011.29.52%E2%80%AFAM.png" alt="Screenshot 2024-09-26 at 11.29.52 AM.png" />


Fixed

**Comment by rbontrager** [comment]
Verified that links are now selectable with the screenreader on IOS.  Verified that per Brea's comments text outside the links are grouped together while the links are being read separately.  Verified that per Lauren's/my comments everything is left aligned when there are multiple lines of words/links.  Verified that the changes did not break android.  Approved by QA.
